pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -676,14 +681,21 @@ func (p *prcHandler) reconcileWorkloadsUsing(ctx context.Context, config string,
 }
 
 func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	helper, err := newProvStoreHelper(c.client)
+	if err != nil {
+		return err
+	}
+
+	c.helper = helper
```

Comment:
Can we avoid mutating the controller here? Related to the other comment about moving the reference of the helper.